transport/server.cc

```diff
@@ -506,8 +506,6 @@ future<foreign_ptr<std::unique_ptr<cql_server::response>>>
                 break;
         }
 
-        tracing::set_username(trace_state, client_state.user());
```

Comment:
I don't think you have a user name decoded up until this point in a general case, do you? 
What if a ` AUTH_RESPONSE` has a Trace bit set?

But except for the point above this patch makes sense.

Reply:
> I don't think you have a user name decoded up until this point in a general case, do you?

`client_state.user()` just returns `optional<auth::authenticated_user>&` so it has to be set way before this call. I did not see that it may be set somewhere in the middle.

> What if a AUTH_RESPONSE has a Trace bit set?

Old code will happely ignore `client_state.set_login(std::move(user))` call that is set much later in the chain.

> However every performance optimization needs to come with benchmarks data proving that there is an improvement. Do you have it?

Of course not :D